src/client/components/pages/RoomAdmin/EditRoomModal.tsx

```diff
@@ -0,0 +1,305 @@
+import React, {
+  FunctionComponent,
+  ReactElement,
+  useState,
+  useRef,
+  useEffect,
+  ChangeEvent,
+  Ref,
+} from 'react';
+import {
+  VARIANT,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Button,
+  ValidationErrorMessage,
+  NoteText,
+  TextInput,
+  POSITION,
+  Form,
+} from 'mark-one';
+import RoomAdminResponse from 'common/dto/room/RoomAdminResponse.dto';
+import ValidationException from 'common/errors/ValidationException';
+import { AxiosError } from 'axios';
+import { LocationAPI } from 'client/api';
+import { ServerErrorInfo } from '../Courses/CourseModal';
+
+interface EditRoomModalProps {
+  /**
+   * Whether or not the modal should be visible on the page.
+   */
+  isVisible: boolean;
+  /**
+   * The current room being edited.
+   */
+  currentRoom: RoomAdminResponse;
+  /**
+   * Handler to be invoked when the edit is successful
+   */
+  onSuccess: () => void;
+  /**
+   * Handler to be invoked when the modal is closed
+   */
+  onClose: () => void;
+}
+
+interface FormErrors {
```

Comment:
Can you add a brief docblock about what this interface represents?

Reply:
Oops, thanks for catching these missing docblocks! I added in docblocks for the various cases through commit [1bf3981](https://github.com/seas-computing/course-planner/pull/581/commits/1bf398187801902dbed320b8e7f7e1bb5e890888).